pkg/logql/log/parser.go

```diff
@@ -486,3 +500,76 @@ func (u *UnpackParser) unpack(it *jsoniter.Iterator, entry []byte, lbs *LabelsBu
 	}
 	return entry, nil
 }
+
+type SyslogParser struct {
+	maxMessageLength int
+	lbs              *LabelsBuilder
+}
+
+func NewSyslogParser() *SyslogParser {
+	return &SyslogParser{maxMessageLength: 1 << 12}
+}
+
+func (l *SyslogParser) Process(line []byte, lbs *LabelsBuilder) ([]byte, bool) {
+	if lbs.ParserLabelHints().NoLabels() {
+		return line, true
+	}
+	l.lbs = lbs
+	b := line[0]
+	r := bytes.NewReader(line)
+	if b == '<' {
+		nontransparent.NewParser(
```

Comment:
In fact I also think the two parser could be part of the struct, if we set a MaxMessageLength high enough ?

Reply:
One of the parser actually doesn't use the length.

Reply:
maxMessageLength: 8192, // size as per RFC5425#section-4.3.1
may be ?

Reply:
I made the two parsers part of the struct.